docs/enhancements/ipxe-host-boot.md

```diff
@@ -0,0 +1,125 @@
+---
+title: ipxe-host-boot
+authors:
+  - "@carbonin"
+  - “@mhrivnak”
+creation-date: 2022-02-22
+last-updated: 2022-02-22
+---
+
+# iPXE Host Boot
+
+## Summary
+
+Not all user infrastructure is able to boot isos directly, and iPXE is a commonly 
+used alternative for booting machines. We should support this by making the 
+application directly serve iPXE artifacts so that systems can boot with no 
+other iPXE infrastructure required.
+
+## Motivation
+
+Booting a host using iPXE should be a first-class operation when using
+assisted-service. This has been found to be at least as common a method 
+as providing the live ISO via virtual media and is also required in some 
+environments. Users are already unpacking the ISO we provide into PXE 
+artifacts and serving those in their infrastructure. We can make this much 
+easier for them by serving those artifacts as well as the iPXE boot script 
+directly from our application.
+
+### Goals
+
+- Serve iPXE artifacts (kernel, rootfs, and initrd)
+- Serve an iPXE boot script
+- Include the ignition in the provided initrd
+
+### Non-Goals
+
+- Support PXE (tftp based)
+- Integration with ironic (separate effort)
```

Comment:
metal3? :)

Reply:
:shrug: I don't know what the difference is honestly.
What I meant here is that this is in support of the ZTP convergence flow rather than implementing any of it.